src/Symfony/Component/ErrorHandler/DebugClassLoader.php

```diff
@@ -320,6 +345,13 @@ public function checkAnnotations(\ReflectionClass $refl, string $class): array
             }
         }
 
+        $parentsAndInterfacesThatHaveMethodsThatReturnsADocType = [];
+        foreach (array_merge($parent ? array_merge([$parent], class_parents($parent, false)) : [], class_implements($class, false)) as $use) {
```

Comment:
can't we use $parentAndOwnInterfaces here?

Reply:
Actually, I'm pretty sure we should use `$parentAndOwnInterfaces`, and that we should skip the notice when a method explicitly declares the `@return` annotation when it parent does also. That would allow the same mechanism as we have for `@param`: when a child class cannot be updated to keep BC, duplicating the annotation next to `@inheritdoc` allows silencing the notice. We're going to need this in Symfony itself :)

Reply:
> Actually, I'm pretty sure we should use $parentAndOwnInterfaces
We can't because we want all implemented interfaces or we can't report implemented methods of parents interfaces (cf "returnTypeGrandParentInterface" test case).

> that we should skip the notice when a method explicitly declares the @return
That is supposed to be the case here. We don't trigger the deprecation if the method has any `@return` PHPDoc annotation (we don't check the type like for `@param`).